src/Microsoft.TestPlatform.Common/Filtering/FilterExpressionWrapper.cs

```diff
@@ -16,26 +18,68 @@ public class FilterExpressionWrapper
         /// <summary>
         /// FilterExpression corresponding to filter criteria
         /// </summary>
-        private FilterExpression filterExpression;
-        
+        private readonly FilterExpression filterExpression;
+
+        /// <remarks>
+        /// Exposed for testing purpose.
+        /// </remarks>
+        internal readonly FastFilter fastFilter;
+
+        private bool UseFilterExpression => this.filterExpression != null;
+
+        private bool UseFastFilter => this.fastFilter != null;
+
         /// <summary>
-        /// Initializes FilterExpressionWrapper with given filterString.  
+        /// Initializes FilterExpressionWrapper with given filterString and options.  
         /// </summary>
-        public FilterExpressionWrapper(string filterString)
+        public FilterExpressionWrapper(string filterString, FilterOptions options = null)
```

Comment:
We can make this a required parameter now?

Reply:
Fixed.